core/types/transaction_marshalling.go

```diff
@@ -334,3 +342,15 @@ func (tx *Transaction) UnmarshalJSON(input []byte) error {
 	// TODO: check hash here?
 	return nil
 }
+
+type depositTxWithNonce struct {
+	DepositTx
+	Nonce uint64
+}
+
+// EncodeRLP ensures that RLP encoding this transaction excludes the nonce. Otherwise, the tx Hash would change
+func (tx *depositTxWithNonce) EncodeRLP(w io.Writer) error {
+	return rlp.Encode(w, tx.DepositTx)
+}
+
+func (tx *depositTxWithNonce) nonce() uint64 { return tx.Nonce }
```

Comment:
I don't think we need to introduce this type. Although it does allow us to report the nonce of a deposit we decode from JSON, we should never actually have to decode a deposit from JSON to show anything to the user. And in the state-transition we don't trust such user-supplied json deposit,